src/include/aerospike/as_operations.h

```diff
@@ -776,35 +776,120 @@ as_operations_add_touch(as_operations* ops);
 AS_EXTERN bool
 as_operations_add_delete(as_operations* ops);
 
+// If you add flags in bits 4 or higher, make sure you make a corresponding
+// change in as_exp_path_modify_flags as well (and vice versa, of course).
+// Only bits 0-3 are unique to select vs. modify operations.
+/**
+ * These flags apply to the as_exp_select_by_path()/
+ * as_operations_select_by_path() functions.
+ */
+typedef enum {
+	/**
+	 * Return a tree from the root (bin) level to the bottom of the tree,
+	 * with only non-filtered out nodes.
+	 */
+	AS_EXP_PATH_SELECT_MATCHING_TREE = 0,
+
+	/**
+	 * Return the list of the values of the nodes finally selected by the context.
+	 * For maps, this returns the value of each (key, value) pair.
+	 */
+	AS_EXP_PATH_SELECT_VALUE = 1,
+
+	/**
+	 * Return the list of the values of the nodes finally selected by the context.
+	 * This is a synonym for AS_EXP_PATH_SELECT_VALUE to make it clear in your
+	 * source code that you're expecting a list.
+	 */
+	AS_EXP_PATH_SELECT_LIST_VALUE = 1,
+
+	/**
+	 * Return the list of map values of the nodes finally selected by the context.
+	 * This is a synonym for AS_EXP_PATH_SELECT_VALUE to make it clear in your
+	 * source code that you're expecting a map.  See also
+	 * AS_EXP_PATH_SELECT_MAP_KEY_VALUE.
+	 */
+	AS_EXP_PATH_SELECT_MAP_VALUE = 1,
+
+	/**
+	 * Return the list of map keys of the nodes finally selected by the context.
+	 */
+	AS_EXP_PATH_SELECT_MAP_KEY = 2,
+
+	/**
+	 * Returns the list of map (key, value) pairs of the nodes finally selected
```

Comment:
When testing this, EXP_PATH_SELECT_MAP_KEY_VALUE returns a list of "flattened" key-value pairs, meaning the list is formatted like `[key1, value1, key2, value2, ...]` where the map looks like `{key1: value1, key2: value2, ...}`. It's not clear in the documentation that this is the expected behavior, so we should mention this